### PR TITLE
Compatibility with networkx>=2.0 and python 3.7+

### PR DESCRIPTION
**Pull request by pylanglois**
networkx supports python 3.8+.
https://networkx.org/documentation/stable/install.html

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -47,10 +47,10 @@ def read(fname):
             'rural_postman_star=postman_problems.examples.star.rpp_star:main'
         ]
     },
-    python_requires='>=2.7',
+    python_requires='>=3.8',
     install_requires=[
         'pandas',
-        'networkx==2.0'
+        'networkx'
```
Comment (brooksandrew) [inline setup.py:53]: Looks like networkx 2.6 supports python 3.7, 3.8, 3.9.  Any reason to restrict to >=3.8 ? 

https://networkx.org/documentation/stable/release/release_2.6.html

File: postman_problems/solver.py [hunk 2]
```diff
@@ -94,7 +94,7 @@ def cpp(edgelist_filename, start_node=None, edge_weight='distance', verbose=Fals
     g_odd_complete = create_complete_graph(odd_node_pairs_shortest_paths, flip_weights=True)
 
     logger_cpp.info('Find min weight matching using blossom algorithm')
-    odd_matching = dedupe_matching(nx.algorithms.max_weight_matching(g_odd_complete, True))
```
Comment (brooksandrew) [inline postman_problems/solver.py:97]: why don't we need the deduping here anymore?

**Review by brooksandrew — COMMENTED** [review]
@pylanglois - thanks for the PR!

Given latest networkx releases cut support for earlier versions of python, I'm generally onboard with this lib following suit.  

It's been a while since new contributions to this lib and my Travis CI has been deactivated on the `.org` domain.  Running into some issues migrating to the new `.com` domain... but once i get that up and tests running again through CI.  will merge this.

Thanks again for helping keep this lib current w the times 👍 


https://networkx.org/documentation/stable/release/release_2.6.html

File: setup.py (hunk 1 above)
Comment (brooksandrew) [reply setup.py:53]: Also if we pin to specific python versions, perhaps also pin networkx with `>=` to the earliest version supported by your changes?  Any reason not to?

**Comment by brooksandrew** [comment]
@pylanglois Closing and re-opening to trigger Travis CI
per https://stackoverflow.com/questions/17606874/trigger-a-travis-ci-rebuild-without-pushing-a-commit

**Comment by brooksandrew** [comment]
Re-opening PR to trigger Travis CI

**Comment by brooksandrew** [comment]
@pylanglois - OK i got in touch with the Travis CI folks and got this project designated as an OSS which qualifies for enough compute credits to turn Travis CI back on to maintain this project. hooray.   Just re-triggered a build on your PR which should now build on each new commit to it.   

Can you access Travis CI to see the logs from each CI run?  Not sure of the permissions for folks that are not me.  If you can get those tests passing (only 2 of 34 failed), I'd like to merge this.  

https://github.com/brooksandrew/postman_problems/pull/68/checks?check_run_id=5646750303

image

File: postman_problems/solver.py (hunk 2 above)
Comment (pylanglois) [reply postman_problems/solver.py:97]: the new version of max_weight_matching returns a set which does not contain duplicate.
https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.matching.max_weight_matching.html

File: setup.py (hunk 1 above)
Comment (pylanglois) [reply setup.py:53]: I usually prefer to support the latest version of all package and the documentation of networkx only contains version 2.7.1 And, I'm not used to pin version but if you want to pin to 2.6 it's ok, I don't really have a good reason not to do it.

**Comment by brooksandrew** [comment]
@pylanglois - are you able to see the Travis CI runs (screenshot below) w details on the tests passing/failing?    Your latest commit looks good as it is passing Python 3.8 and 3.9.  It looks like some weird dependency issue with 3.7 and coveralls.  And 3.10 looks like it might not be available on Travis.  So I'd be OK to just comment out 3.7 and 3.10 in the travis configuration for now.  And then we can merge your upgrade in.


image

**Comment by brooksandrew** [comment]
hi @pylanglois - just following up.  Per the last comment, I'd be OK to test this on 3.8 and 3.9 only for now in Travis CI.  If that passes, lets merge this.

**Comment by pylanglois** [comment]
Ok. I started a little experiment with github actions. I will close that shortly and go with your suggestion.

**Comment by pylanglois** [comment]
hi @brooksandrew,

The tests I ran seems to be ok for all python versions above 3.7.1. See: https://github.com/pylanglois/postman_problems/actions/runs/2204057017

Do I put `python_requires='>=3.7.1'` as requirement?

**Review by brooksandrew — APPROVED** [review]
LGTM!  Thanks for the ping on this.  Saw it when on mobile when i was away from computer and forgot to return to it.  

And double thanks for verifying that the test suite runs on 3.7 and 3.10.  I'm not sure what's going on with Travis CI machines, but that's good to know that most users should be fine with >3.7, so I'm good with the python version as you have it.

**Pull request merged**